Introduce `ignore_missing` argument to A, AAAA and CNAME data

sergei-ivanov opened the pull request:
Fixes #75

bodgit commented:
Personally, I'd rather see this option added to all of the datasources rather than only add it to some of them, otherwise they become inconsistent with each other.

Also, (and I'm not sure if this is possible), you are potentially masking any type of DNS error behind this boolean rather than specifically just NXDOMAIN.

sergei-ivanov commented:
I was about to do it for all data sources, but had to stop myself, because I wanted to hear some feedback on the proposed changes (much appreciated!). I'll be happy to introduce the same logic for all remaining record types.

I have also spent some time looking at the underlying library implementation, but I could not figure out a clean and reliable way to distinguish "no such host" failures from other errors. I looked at it earlier, and I looked at it again just now. Still cannot figure it out.

So may be in this context `ignore_missing` is a slightly misleading name, and I'll appreciate ideas here. I thought at some point to invert that flag and call it `fail_on_error`, with `true` being the default (to preserve the current behaviour).

sergei-ivanov commented:
Sorry, it took a few months to get back to this. I have rebased on the latest master, renamed the argument to `ignore_errors` and added it to all data sources. Documentation has been updated accordingly.

sergei-ivanov commented:
I'll fix the failing test tonight.

sergei-ivanov commented:
Looks like the DNS PTR record has been updated on Google side:
```
$ dig -x 8.8.8.8 -t PTR

;; ANSWER SECTION:
8.8.8.8.in-addr.arpa.	68365	IN	PTR	dns.google.

```

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/terraform-provider-dns?pullRequest=76)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/terraform-provider-dns?pullRequest=76)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)


---
**1** out of **2** committers have signed the CLA.

- [x] pselle
- [ ] sergei-ivanov

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/terraform-provider-dns?pullRequest=76).</sub>

freakinhippie commented:
This seems silly. The `hashicorp-cla` bot wants the CLA signed by a user that merged `master` into the feature branch. Perhaps it should have been a rebase rather than a merge but it doesn't make much sense to require the CLA for that.

I any case, this feature would be very helpful for me. I'd be willing to put some effort into getting this up-to-date if that's acceptable.

sergei-ivanov commented:
The provider has moved on a lot since the original PR and especially the tests were relocated and substantially refactored. I started rebasing, but it quickly became a major chore, and I had to put it on hold because of a lack of spare time on my part. I'll try to get back to it, but I need some encouragement from the provider team, because I don't want to waste my effort on something that might be ignored.

atrull commented:
I'd like this one too @sergei-ivanov #sergeistandard <B see my overbuilt myip module, where I want more survivability on silly things.
